x-pack/elastic-agent/magefile.go

```diff
@@ -579,6 +579,10 @@ func packageAgent(requiredPackages []string, packagingFn func()) {
 		}
 
 		os.Setenv(agentDropPath, dropPath)
+		if runtime.GOARCH == "arm64" {
+			os.Setenv("PLATFORMS", runtime.GOOS+"/"+runtime.GOARCH)
+			defer os.Unsetenv("PLATFORMS")
```

Comment:
```suggestion
			defer os.Unsetenv("PLATFORMS")
			os.Setenv("PLATFORMS", runtime.GOOS+"/"+runtime.GOARCH)
```

Reply:
it is a deferred unset, it would happen at the end of the function.